internal/datasources/web.go

```diff
@@ -0,0 +1,57 @@
+package datasources
+
+import (
+	"bytes"
+	"io"
+	"mime"
+	"net/http"
+)
+
+type WebFileDatasource struct {
+	uri string
+}
+
+func NewWebFileDatasource(uri string) *WebFileDatasource {
+	return &WebFileDatasource{uri}
+}
+
+func (ds *WebFileDatasource) Load() (map[string]any, error) {
+	data := make(map[string]any)
+
+	res, err := http.Get(ds.uri)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	ct := res.Header.Get("Content-Type")
+	mt, _, _ := mime.ParseMediaType(ct)
+
+	b, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+	r := bytes.NewReader(b)
+
+	if mt == "application/json" {
```

Comment:
this whole section is repeated, move it to be outside the conditions, and have the conditions only initialize the `ds` variable
```go
data, err = ds.Load()
if err != nil {
	return nil, err
}
```